### PR TITLE
[Parser] Parse throw_ref

### PR DESCRIPTION
**Pull request by tlively**

### CODE REVIEW

**Comment by tlively** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/WebAssembly/binaryen/6238?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `main`
  * **PR #6237** <a href="https://app.graphite.dev/github/pr/WebAssembly/binaryen/6237?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #6238** <a href="https://app.graphite.dev/github/pr/WebAssembly/binaryen/6238?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/WebAssembly/binaryen/6238?utm_source=stack-comment).

**Review by kripken — APPROVED** [review]

File: src/wasm.h [hunk 1]
```diff
@@ -1522,6 +1522,7 @@ class Rethrow : public SpecificExpression<Expression::RethrowId> {
 // 'throw_ref' from the new EH proposal
 class ThrowRef : public SpecificExpression<Expression::ThrowRefId> {
 public:
+  ThrowRef() {}
```
Comment (aheejin) [inline src/wasm.h:1525]:
```suggestion
  ThrowRef() = default;
```
Nit: It looks other classes in this file use this

**Review by aheejin — APPROVED** [review]
LGTM % comments

File: src/parser/contexts.h [hunk 2]
```diff
@@ -1406,7 +1407,9 @@ struct ParseDefsCtx : TypeParserCtx<ParseDefsCtx> {
       irBuilder.makeTry(label ? *label : Name{}, type.getSignature().results));
   }
 
-  Result<> makeTryTable(Index pos, std::optional<Name> label, HeapType type,
+  Result<> makeTryTable(Index pos,
```
[the remark this answers is not in the document]
Comment (tlively) [reply src/parser/contexts.h:1410]: oops, yep

**Comment by tlively** [comment]
### Merge activity

* **Jan 25, 3:32 PM**: @tlively started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/WebAssembly/binaryen/6238).

**Pull request merged**